src/test/java/com/github/dariobalinzo/task/ElasticSourceTaskTest.java

```diff
@@ -339,8 +392,7 @@ public void shouldRunSourceTaskWhitelist() throws IOException, InterruptedExcept
         //when (fetching first page)
         task.start(conf);
         List<SourceRecord> poll1 = task.poll();
-        assertEquals("Struct{fullName=Test}", poll1.get(0).value().toString());
-
+        assertEquals(((Struct) poll1.get(0).value()).get("fullName"), "Test");
         task.stop();
```

Comment:
may you please assert here that poll1.get(0).value() size is 1 (to be sure that the whitelist filter is working and emitting only the fullname field, with no additional fields).